storm-core/src/jvm/backtype/storm/blobstore/BlobStoreAclHandler.java

```diff
@@ -0,0 +1,387 @@
+        //ignored
+      } else {
+        throw new IllegalArgumentException("");
+      }
+    }
+    return ret;
+  }
+
+  public static AccessControl parseAccessControl(String str) {
+    String[] parts = str.split(":");
+    String type = "other";
+    String name = "";
+    String access = "-";
+    if (parts.length > 3) {
+      throw new IllegalArgumentException("Don't know how to parse "+str+" into an ACL value");
+    } else if (parts.length == 1) {
+      type = "other";
+      name = "";
+      access = parts[0];
+    } else if (parts.length == 2) {
+      type = "user";
+      name = parts[0];
+      access = parts[1];
+    } else if (parts.length == 3) {
+      type = parts[0];
+      name = parts[1];
+      access = parts[2];
+    }
+    AccessControl ret = new AccessControl();
+    ret.set_type(parseACLType(type));
+    ret.set_name(name);
+    ret.set_access(parseAccess(access));
+    return ret;
+  }
+
+  private static String accessToString(int access) {
+    StringBuffer ret = new StringBuffer();
+    ret.append(((access & READ) > 0) ? "r" : "-");
+    ret.append(((access & WRITE) > 0) ? "w" : "-");
+    ret.append(((access & ADMIN) > 0) ? "a" : "-");
+    return ret.toString();
+  }
+
+  public static String accessControlToString(AccessControl ac) {
+    StringBuffer ret = new StringBuffer();
+    switch(ac.get_type()) {
+      case OTHER:
+        ret.append("o");
+        break;
+      case USER:
+        ret.append("u");
+        break;
+      default:
+        throw new IllegalArgumentException("Ahh don't know what a type of "+ac.get_type()+" means ");
+    }
+    ret.append(":");
+    if (ac.is_set_name()) {
+      ret.append(ac.get_name());
+    }
+    ret.append(":");
+    ret.append(accessToString(ac.get_access()));
+    return ret.toString();
+  }
+
+  public static void validateSettableACLs(String key, List<AccessControl> acls) throws AuthorizationException {
+    Set<String> aclUsers = new HashSet<>();
+    List<String> duplicateUsers = new ArrayList<>();
+    for (AccessControl acl : acls) {
+      String aclUser = acl.get_name();
+      if (aclUser != null && !aclUser.isEmpty() && !aclUsers.add(aclUser)) {
+        LOG.error("'{}' user can't appear more than once in the ACLs", aclUser);
+        duplicateUsers.add(aclUser);
+      }
+    }
+    if (duplicateUsers.size() > 0) {
+      String errorMessage  = "user " + Arrays.toString(duplicateUsers.toArray())
+          + " can't appear more than once in the ACLs for key [" + key +"].";
+      throw new AuthorizationException(errorMessage);
+    }
+  }
+
+  private Set<String> constructUserFromPrincipals(Subject who) {
+    Set<String> user = new HashSet<String>();
+    if (who == null) {
+      LOG.debug("in validate acl who is null");
+    } else {
+      LOG.debug("in validate acl: " + who);
+    }
+    if (who != null) {
+      for (Principal p : who.getPrincipals()) {
+        user.add(_ptol.toLocal(p));
+      }
+    }
+    return user;
+  }
```

Comment:
We could just have an `else` where we `LOG.debug("in validate acl: {}", who);`
The current code will run even when we aren't logging at DEBUG level.